tools/data_faker/data_faker/statistics.py

```diff
@@ -177,8 +177,8 @@ def create_fake_data(self, cursor: DBCursor, args: argparse.Namespace) -> None:
                     category=BalanceType.ASSET,
                     time=from_ts,
                     asset=assets[idx],
-                    amount=str(new_amount),
-                    usd_value=str(action(FVal(assets_data[idx].usd_value), value)),
+                    amount=FVal(new_amount),
+                    usd_value=FVal(action(FVal(assets_data[idx].usd_value), value)),
```

Comment:
Again ... please read the code you are changing. To me it really seems as if you don't understand what you are doing and are just mechanically pasting stuff around.

1. You don't need to cast the whole thing to FVal since Fval + Fval -> Fval and Fval - Fval -> Fval
2. And thanks to your change `usd_value` of a `DBAssetBalance` is now an FVal
```suggestion
                    usd_value=action(assets_data[idx].usd_value, value),
```